engine/src/test/java/io/camunda/zeebe/engine/processing/resource/ResourceDeletionMultiPartitionTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.resource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.intent.CommandDistributionIntent;
+import io.camunda.zeebe.protocol.record.intent.DecisionIntent;
+import io.camunda.zeebe.protocol.record.intent.DecisionRequirementsIntent;
+import io.camunda.zeebe.protocol.record.intent.ResourceDeletionIntent;
+import io.camunda.zeebe.protocol.record.value.CommandDistributionRecordValue;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestWatcher;
+
+public class ResourceDeletionMultiPartitionTest {
+
+  private static final int PARTITION_COUNT = 3;
+  private static final String DMN_RESOURCE = "/dmn/decision-table.dmn";
+
+  @Rule public final EngineRule engine = EngineRule.multiplePartition(PARTITION_COUNT);
+  @Rule public final TestWatcher watcher = new RecordingExporterTestWatcher();
+
+  @Test
+  public void shouldTestDmnLifecycle() {
+    // given
+    engine.deployment().withXmlClasspathResource(DMN_RESOURCE).deploy();
+
+    final var resourceKey =
+        RecordingExporter.decisionRequirementsRecords()
+            .withIntent(DecisionRequirementsIntent.CREATED)
+            .withDecisionRequirementsId("force_users")
+            .getFirst()
+            .getKey();
+
+    // when
+    engine.resourceDeletion().withResourceKey(resourceKey).delete();
+
+    // then
+    final var index = new AtomicInteger();
+    assertThat(
+            RecordingExporter.records()
+                .withPartitionId(1)
+                .limit(
+                    r ->
+                        r.getIntent().equals(ResourceDeletionIntent.DELETED)
+                            && index.incrementAndGet() == 3))
```

Comment:
🔧 As discussed this limit is somewhat unclear (I had to look up incrementAndGet to make sure it does what I expect). It would be great to move this into a `limitByCount(Predicate, int)` method.